metrics/collectors/economic.go

```diff
@@ -116,28 +131,46 @@ func (ec *EconomicCollector) Collect(ch chan<- prometheus.Metric) {
 		ch <- prometheus.MustNewConstMetric(ec.slashingStatus, prometheus.CounterValue, operatorIsFrozenFloat)
 	}
 
-	// collect delegatedShares metric
-	for _, strategyAddr := range ec.strategyAddrs {
-		sharesWei, err := ec.elReader.GetOperatorSharesInStrategy(context.Background(), ec.operatorAddr, strategyAddr)
-		if err != nil {
-			ec.logger.Error("Failed to get shares", "err", err)
-		} else {
-			// We'll emit all 3 units in case this is needed for whatever reason
-			// might want to change this behavior if this is emitting too many metrics
-			sharesWeiFloat, _ := sharesWei.Float64()
-			// TODO(samlaf): add the token name.. probably need to have a hardcoded dict per env (mainnet, goerli, etc)? Is it really that important..?
-			ch <- prometheus.MustNewConstMetric(ec.delegatedShares, prometheus.GaugeValue, sharesWeiFloat, strategyAddr.String(), "wei", "token")
-
-			sharesGweiFloat, _ := sharesWei.Div(sharesWei, big.NewInt(1e9)).Float64()
-			ch <- prometheus.MustNewConstMetric(ec.delegatedShares, prometheus.GaugeValue, sharesGweiFloat, strategyAddr.String(), "gwei", "token")
-
-			sharesEtherFloat, _ := sharesWei.Div(sharesWei, big.NewInt(1e18)).Float64()
-			ch <- prometheus.MustNewConstMetric(ec.delegatedShares, prometheus.GaugeValue, sharesEtherFloat, strategyAddr.String(), "ether", "token")
-		}
-	}
-
 	// collect registeredStake metric
 	// TODO(samlaf): implement this. probably have to call the BLSOperatorStateRetriever contract?
 	// probably should start using the avsregistry service instead of chainio clients so that we can
 	// swap out backend for a subgraph eventually
+	quorumNums, blsOperatorStateRetrieverOperator, err := ec.avsRegistryReader.GetOperatorsStakeInQuorumsOfOperatorAtCurrentBlock(context.Background(), ec.operatorId)
+	if err != nil {
+		ec.logger.Error("Failed to get operator stake", "err", err)
+	} else {
+		for quorumIdx, quorumNum := range quorumNums {
+			// TODO: this is stupid.. when AVSs scale to have 5K operators we'll be running through a bunch of operators
```

Comment:
what is the reason of not doing this right now?

Reply:
Painful man.. I'm tired of implementing this crap, plus we only have 200 operators max so shouldn't be that big of a deal. If you force me to I can do it tomorrow :(